### PR TITLE
Add properties files to configure startup and installation scripts

### PR DESCRIPTION
**Pull request by tlrx**
Many scripts are used to start/stop and install/uninstall elasticsearch. These scripts share a lot of configuration properties like directory paths, max value for a setting, default user etc. Most of the values are identical but some of them are different depending of the platform (Debian-based or Redhat-based OS), depending of the way elasticsearch is started (shell script, systemd, sysv-init...) or the way it is installed (zip, rpm, deb...). Today the values are duplicated in multiple places, making it difficult to maintain the scripts or to update a value.

This pull request make this more uniform: values used in scripts must be defined in a common `packaging.properties` file. Each value can be overridden in another specific `packaging.properties` file for Debian or Redhat. All startup and installation scripts are filtered with the common then the custom `packaging.properties` files before being packaged as a zip/tar.gz/rpm/dpkf archive.

Also, this functionality will be very helpful when mutualizing scripts between rpm and debian packages (work in progress in another pull request)

### CODE REVIEW

**Comment by tlrx** [comment]
@spinscale we talked about this so I assign it to you :) I'd be glad if you have time to review it, thanks!

File: pom.xml [hunk 1]
```diff
@@ -358,6 +368,10 @@
 
     <build>
 
+        <filters>
```
Comment (spinscale) [inline pom.xml]: maybe add a one-line comment here, what this paragraph does? Not obvious for all of the readers

File: src/packaging/common/packaging.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+#
+# Properties defined here can be overridden with specific settings,
+# like in rpm/packaging.properties and deb/packaging.properties.
+
+# Default values for min/max heap memory allocated to elasticsearch java process
+packaging.elasticsearch.heap.min=256m
+packaging.elasticsearch.heap.max=1g
+
+# Simple marker to check that properties are correctly overridden
+packaging.type=tar.gz,gzip
```
Comment (spinscale) [inline src/packaging/common/packaging.properties]: this marker seems never to be used, read or accessed? Is it needed? If so, lets try to find a better description?

Comment (tlrx) [inline src/packaging/common/packaging.properties]: This marker is useful to check that a file has been filtered with the appropriate packaging.properties file. It's possible that I'll add a comment like this in some files:
```
# Built for ${project.name}-${project.version} (${packaging.type})
```
But you're right, for now it is not used. I can removed it.

**Comment by electrical** [comment]
Did a test with the puppet module and ran into an issue at package removal of the debian package.

```
rmdir: failed to remove `/var/lib/elasticsearch': No such file or directory
```

**Comment by tlrx** [comment]
@electrical I don't think this is due to the changes in this PR. Do you have the same message with a RPM built from master?

**Comment by electrical** [comment]
Rpm package works fine.
Don't have the issue with the current debian packages.

**Comment by electrical** [comment]
LGTM. error is gone.

**Pull request merged**